lib/improver/utilities/spatial.py

```diff
@@ -0,0 +1,159 @@
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+""" Provides support utilities."""
+
+import copy
+from iris.coords import CellMethod, DimCoord
+from iris.cube import Cube
+from iris.exceptions import InvalidCubeError
+import numpy as np
+
+
+class DiscreteDifferenceBetweenAdjacentGridSquares(object):
+
+    """
+    Calculate the discrete difference between adjacent grid squares within
+    a cube. The discrete difference is calculated along the x and y axis
+    individually.
+    """
+
+    def __init__(self):
+        """
+        Initialise class.
+        """
+        pass
+
+    @staticmethod
+    def create_discrete_difference_cube(
+            cube, coord_name, diffs_along_axis):
+        """
+        Put the discrete difference array into a cube with the appropriate
+        metadata.
+
+        Parameters
+        ----------
+        cube : Iris.cube.Cube
+            Cube from which the discrete differences will be calculated.
+        coord_name : String
+            The name of the coordinate over which the discrete difference
+            will be calculated.
+        diffs_along_axis : numpy array
+            Array containing the discrete differences.
+
+        Returns
+        -------
+        diff_cube : Iris.cube.Cube
+            Cube after the discrete differences have been calculated along the
+            specified axis.
+        """
+        points = cube.coord(coord_name).points
+        mean_points = (points[1:] + points[:-1]) / 2
+
+        # Copy cube metadata and coordinates into a new cube.
+        # Create a new coordinate for the coordinate along which the discrete
+        # difference has been calculated.
+        metadata_dict = copy.deepcopy(cube.metadata._asdict())
+        diff_cube = Cube(diffs_along_axis, **metadata_dict)
+
+        for coord in cube.dim_coords:
+            dims = cube.coord_dims(coord)
+            if coord.name() in [coord_name]:
+                coord = DimCoord(
+                    mean_points, standard_name=coord.standard_name,
+                    units=coord.units)
+            diff_cube.add_dim_coord(coord.copy(), dims)
+        for coord in cube.aux_coords:
+            dims = cube.coord_dims(coord)
+            diff_cube.add_aux_coord(coord.copy(), dims)
+        for coord in cube.derived_coords:
+            dims = cube.coord_dims(coord)
+            diff_cube.add_aux_coord(coord.copy(), dims)
+
+        # Add metadata to indicate that a discrete difference has been
+        # calculated.
+        cell_method = CellMethod("discrete_difference", coords=[coord_name])
+        diff_cube.add_cell_method(cell_method)
+        diff_cube.attributes["direction_of_discrete_difference"] = (
+            "forward_difference")
```

Comment:
I think we should change it to form_of_discrete_difference too. That might be  slightly less confusing?